sagemaker_batch_transform/tensorflow_open-images_tfrecord/tensorflow-serving-tfrecord-python-sdk.ipynb

```diff
@@ -0,0 +1,385 @@
+    "\n",
+    "For improved performance, we specify two additional parameters `max_concurrent_transforms` and `max_payload`, which control the maximum number of parallel requests that can be sent to each instance in a transform job at a time, and the maximum size of each request body.\n",
+    "\n",
+    "When performing inference on entire S3 objects that cannot be split by newline characters, such as images, it is recommended that you set `max_payload` to be slightly larger than the largest S3 object in your dataset, and that you experiment with the `max_concurrent_transforms` parameter in powers of two to find a value that maximizes throughput for your model. For example, we’ve set `max_concurrent_transforms` to 64 after experimenting with powers of two, and we set `max_payload` to 1, since the largest object in our S3 input is less than one megabyte."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "output_path = os.path.join(s3_path, 'output')\n",
+    "\n",
+    "env = {'SAGEMAKER_TFS_ENABLE_BATCHING': 'true',\n",
+    "      'SAGEMAKER_TFS_BATCH_TIMEOUT_MICROS': '50000',\n",
+    "      'SAGEMAKER_TFS_MAX_BATCH_SIZE': '16'}\n",
+    "tensorflow_serving_transformer = tensorflow_serving_model.transformer(\n",
+    "                                     instance_count=2,\n",
+    "                                     strategy='SingleRecord',\n",
+    "                                     instance_type='ml.p3.2xlarge',\n",
+    "                                     max_concurrent_transforms=64,\n",
+    "                                     max_payload=1,\n",
+    "                                     output_path=output_path,\n",
+    "                                     env=env,\n",
+    "                                     assemble_with='Line')\n",
+    "\n",
+    "print('Transform input S3 path:  {}'.format(input_path))\n",
+    "print('Transform output S3 path: {}'.format(output_path))\n",
+    "tensorflow_serving_transformer.transform(input_path, split_type='TFRecord', content_type='application/x-tfexample')\n",
+    "tensorflow_serving_transformer.wait()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In addition to performance parameters, we specify AssembleWith to be “Line”, to instruct our Transform Job to assemble the individual predictions in each object by newline characters rather than concatenating them.\n",
+    "\n",
+    "Furthermore, we specify certain environment variables, which are passed to the TensorFlow Serving Container, and are used to enable request batching, a TensorFlow Serving feature that allows records from multiple requests be batched together. When carefully configured, this can improve throughput, especially with GPU-accelerated inference."
```

Comment:
Can we link to the full set of environment variables supported by the TensorFlow Serving container for batching?

Reply:
Sure, done